src/test/javascript/spec/component/competencies/competency-form.component.spec.ts

```diff
@@ -95,6 +95,7 @@ describe('CompetencyFormComponent', () => {
         competencyFormComponentFixture.whenStable().then(() => {
             expect(submitFormSpy).toHaveBeenCalledOnce();
             expect(submitFormEventSpy).toHaveBeenCalledOnce();
+            discardPeriodicTasks();
         });
```

Comment:
Same here, I don't think this is being executed. Do we need `whenStable` at all?

```suggestion
        flush();
        expect(submitFormSpy).toHaveBeenCalledOnce();
        expect(submitFormEventSpy).toHaveBeenCalledOnce();
        discardPeriodicTasks();
```